### PR TITLE
chore: add loop to find closest target's fee estimate

### PR DESCRIPTION
**Pull request by slavastartsev**
Closes https://github.com/bob-collective/bob/issues/364

### CODE REVIEW

File: sdk/src/esplora.ts [hunk 1]
```diff
@@ -333,7 +333,11 @@ export class EsploraClient {
      */
     async getFeeEstimate(confirmationTarget: number): Promise<number> {
         const response = await this.getJson<Record<number, number>>(`${this.basePath}/fee-estimates`);
-        return response[confirmationTarget];
+        if (response[confirmationTarget]) return response[confirmationTarget];
+        const TARGET_MAX = 1008;
+        for (let target = confirmationTarget + 1; target <= TARGET_MAX; target++) {
+            if (response[target]) return response[target];
+        }
```
Comment (gregdhill) [inline sdk/src/esplora.ts:340]: I think a better approach is just to convert the `Record` to an array, sort by the confirmation number, and return the first entry in that array.

**Review by gregdhill — CHANGES_REQUESTED** [review]

File: sdk/src/esplora.ts (hunk 1 above)
Comment (slavastartsev) [reply sdk/src/esplora.ts:340]: Speaking of time/memory complexity this approach will take O(n) memory + O(n logn) time
Simple for loop takes just O(n) time

Comment (slavastartsev) [reply sdk/src/esplora.ts]: Fixed. Desc sorting + pop()

Comment (gregdhill) [reply sdk/src/esplora.ts]: You were right with the complexity in your comment before btw ;)

Comment (gregdhill) [reply sdk/src/esplora.ts]: but actually I'm not so sure we want this need to consider, fee rate on testnet should always be 1

**Comment by gregdhill** [comment]
Not sure we should fix it like this actually, I think the method should throw an error if the confirmation target isn't available. We are also trying to get around testnet being broken, if we change anything it should be in the UI or sats-wagmi - maybe hardcode the value to 1 instead in that case as it should be.

**Pull request closed without merging**